src/v/redpanda/application.cc

```diff
@@ -713,16 +716,45 @@ void application::wire_up_redpanda_services() {
           std::ref(metadata_cache),
           std::ref(partition_manager));
         coprocessing->start().get();
+    }
 
+    if (v8_enabled()) {
+        syschecks::systemd_message("Creating v8::api").get();
+        const auto& cfg = config::shard_local_cfg();
+        construct_single_service(
+          _v8_engine,
+          ss::engine().alien(),
+          cfg.core_for_executor,
+          cfg.executor_queue_size);
+        _v8_engine->start().get();
+    }
+
+    if (coproc_enabled() || v8_enabled()) {
+        syschecks::systemd_message(
+          "Creating event listener for coproc internal topic")
+          .get();
         construct_single_service(_coproc_event_listener);
 
-        _async_event_handler
-          = std::make_unique<coproc::wasm::async_event_handler>(
-            _coproc_event_listener->get_abort_source(),
-            coprocessing->get_pacemaker());
+        if (coproc_enabled()) {
```

Comment:
Too much code is already in `application.cc` why do we want to move this stuff out into the application? Could what you need just live in `v8::api`?

Reply:
Maybe, but I nned to provide it to data_policy_manager and need in all situation (when v8_enable on not)